### PR TITLE
Feat/handle abbrev

### PR DESCRIPTION
**Pull request by PiotrekZak**
Improved coverage and accuracy of geocoder

### CODE REVIEW

**Comment by PiotrekZak** [comment]
feat/handle_abbrev is an extension of feat/data_reduction

feat/handle_abbrev contains all changes made in feat/data_reduction

File: zrp/prepare/prepare.py [hunk 1]
```diff
@@ -57,7 +57,6 @@ def transform(self, input_data):
             Dataframe to be transformed
```
Comment (kasey-zest) [inline zrp/prepare/prepare.py:57]: Docstring incomplete. Please add a description of the transform (ie "Transform the data")

Comment (kasey-zest) [inline zrp/prepare/prepare.py:57]: Incorrect parameter provided in the fit

File: zrp/prepare/preprocessing.py [hunk 2]
```diff
@@ -108,50 +108,50 @@ def reduce_whitespace(data):
     return data.apply(lambda x: x.str.strip().str.replace(" +", " ", regex=True))
 
 
-def replicate_address(data, i, street_suffix_mapping, unit_mapping):
-    """
-    Replicate street addresses 
+# def replicate_address(data, i, street_suffix_mapping, unit_mapping):
```
Comment (kasey-zest) [inline zrp/prepare/preprocessing.py:111]: please remove instead of commenting out

**Review by kasey-zest — CHANGES_REQUESTED** [review]
please resolve comments

File: zrp/prepare/prepare.py (hunk 1 above)
Comment (PiotrekZak) [reply zrp/prepare/prepare.py:57]: Parameter data is used in lines 38 and 46. Changed the parameter name to input_data so the name is the same as in transform method.

File: zrp/prepare/preprocessing.py (hunk 2 above)
Comment (PiotrekZak) [reply zrp/prepare/preprocessing.py:111]: Done

File: zrp/prepare/prepare.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (PiotrekZak) [reply zrp/prepare/prepare.py:57]: done

File: zrp/prepare/preprocessing.py [hunk 3]
```diff
@@ -204,8 +204,30 @@ def transform(self, data):
...
     
-def replicate_address_2(data, street_address, street_suffix_mapping, unit_mapping):
+    Parameters
+    ----------
+    data: pd.DataFrame
+        DataFrame to make changes to 
+    house_number: str
+        Name of street address column
+    add_to_flg: str
+        Modification to replicate_flg
```
[the remark this answers is not in the document]
Comment (PiotrekZak) [reply zrp/prepare/preprocessing.py:218]: Changed the parameter description to:
    add_to_flg: int
        'replicate_flg' column indicates the order of preference of replicated rows. The smaller the flag the higher the preference. 'add_to_flg' value is added to the flag of replicated rows.

File: zrp/prepare/preprocessing.py [hunk 4]
```diff
@@ -600,15 +672,17 @@ def transform(self, data, processed, replicate=False):
...
             print("      ...replicating address")
-            data = replicate_address_2(data, self.street_address, street_suffix_mapping, unit_mapping)
-            data = replicate_house_number(data, self.house_number)
-        else: 
-            data = data.reset_index(drop=False)
-        
+            data = replicate_house_number(data, self.house_number, 1)
+            data = replicate_address_2(data, self.street_address, street_suffix_mapping, 10)
+            data = replicate_north_n(data, self.street_address, 100)
+#             data = replicate_address_3(data, self.street_address, street_suffix_mapping_new_only, 1000)
```
[the remark this answers is not in the document]
Comment (PiotrekZak) [reply zrp/prepare/preprocessing.py:682]: done

File: zrp/prepare/preprocessing.py [hunk 5]
```diff
@@ -236,40 +258,90 @@ def replicate_address_2(data, street_address, street_suffix_mapping, unit_mappin
...
+                        '\\bSUDOESTE\\b' : 'SO'}
+
+    df_base =  data.copy()
+    data['replicate_flg'] = data['replicate_flg'].apply(lambda s: str(int(s) + add_to_flg).zfill(6))
+    data[street_address] = data[street_address].replace(north_n_mapping, regex=True)
+    dataout = pd.concat([df_base, data], axis=0)
+    dataout = dataout.drop_duplicates(keep = 'first', subset = [col for col in dataout.columns if col != 'replicate_flg'])
+    return dataout
+
+# def replicate_n_north(data, street_address, add_to_flg):
```
[the remark this answers is not in the document]
Comment (PiotrekZak) [reply zrp/prepare/preprocessing.py:311]: done

File: zrp/prepare/geo_lookup.py [hunk 6]
```diff
@@ -319,3 +319,90 @@ def transform(self, st_cty_code, save_table=True):
...
+    """
+    This class loops through GeoLookUpBuilder() to build geo lookup tables
+    
+    Parameters
+    ----------
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.st_dict = dict()
```
[the remark this answers is not in the document]
Comment (PiotrekZak) [reply zrp/prepare/geo_lookup.py:333]: Deleted from init

File: zrp/prepare/geo_lookup.py [hunk 7]
```diff
@@ -319,3 +319,90 @@ def transform(self, st_cty_code, save_table=True):
...
+
+        Parameters
+        ----------
+        """       
+        try:
+            geo_build.transform(st_cty_code, save_table = save_table)
+        except: 
+            pass
+    
+    def transform(self, st_list = None, save_st_cty_tables=True, save_st_tables=True, n_jobs = -1):
```
[the remark this answers is not in the document]
Comment (PiotrekZak) [reply zrp/prepare/geo_lookup.py:366]: done

**Review by kasey-zest — APPROVED** [review]

**Pull request merged**